src/core/infrastructure/fetch.js

```diff
@@ -0,0 +1,29 @@
+import {isFunction, noop} from 'core/services/utility';
+
+export default class Fetch {
+	constructor(source) {
+		this.source = source;
+		this.busy = null;
+		this.result = null;
+	}
+
+	run() {
+		const fetch = isFunction(this.source) ? this.source(...arguments) : this.source;
```

Comment:
const p = new Promise()
const fecth = this.source(...arguments, {resolve: () => Promise,.resolve(p), reject: Promise.reject(p});